client/app/main/visualization/visualization.controller.js

```diff
@@ -1,144 +1,123 @@
 'use strict';
 
-function VisualizationCtrl($scope, $timeout, Sidebar, Visualization) {
+var defaultChartConfig, categorySettings, dataToSeries;
+
+function VisualizationCtrl($scope, $timeout, School, Sidebar, Visualization) {
   $scope.sidebar = Sidebar;
-  $scope.studentType = 'nonCfa';
-  $scope.data = { 
-    cfa: { 
-      arca: {
-        count: 0, pct: 0
-      }, 
-      ca: {
-        count: 0, pct: 0
-      }, 
-      other: {
-        count: 0, pct: 0
-      } 
-    }, 
-    nonCfa: { 
-      arca: {
-        count: 0, pct: 0
-      }, 
-      ca: {
-        count: 0, pct: 0
-      }, 
-      other: {
-        count: 0, pct: 0
-      } 
-    } 
-  };
-  $scope.chartConfig = {
-    options: {
-      chart: {
-        plotBackgroundColor: null,
-        plotBorderWidth: null,
-        plotShadow: false,
-        type: 'pie'
-      },
-      title: {
-        text: 'ARCA vs CA'
-      },
-      tooltip: {
-        pointFormat: '{series.name}<br><b>{point.percentage:.1f}%</b><br><em>{point.count} tot.</em>'
-      },
-      plotOptions: {
-        pie: {
-          allowPointSelect: true,
-          cursor: 'pointer',
-          dataLabels: {
-            enabled: false,
-            format: '<b>{point.name}</b>: {point.percentage:.1f} %',
-          },
-          showInLegend: true
-        }
-      }
-    },
+  $scope.selected = {};
+  School.query({controller: 'names'}, function(schools) {
+    if (schools.length > 1) {
+      schools.push({_id: 'combined', name: 'Combined'});
+    }
+    $scope.schools = schools;
+  });
+
+  $scope.cfaConfig = _.merge({
+    options: {title: {text: 'Enrolled in Child First Authority'}},
     func: function(chart) {
-      $scope.chartObj = chart;
+      $scope.cfaChartObj = chart;
     }
```

Comment:
This is an easy way to make the layout responsive. The other option is to keep a watch on the width of the parent object and update the chart options and draw on resize.